src/screens/SendSheet.js

```diff
@@ -317,11 +328,14 @@ export default function SendSheet(props) {
   ]);
 
   useEffect(() => {
-    if (currentProvider) {
+    const currentProviderNetwork = ethereumUtils.getNetworkFromChainId(
+      currentProvider?._network.chainId
+    );
+    if (currentProviderNetwork === currentNetwork) {
```

Comment:
When this wouldn't be true?
Based on the previous useEffect witb the switch both should be always in sync

Reply:
there is a lag between, could be due to the updateNeworkAndProvider bring async, without this change the balances get queried too early and return null